Zapper-fi/studio
# feat(rigoblock): optimize underlying token definitions

## Description


This PR optimizes the Rigoblock app. The token list is retrieved just once in getDefinitions and stored in each definition. Only tokens with positive balances are returned, improving the efficiency and readability of held tokens display. The code is also modified to remove the need for a token blacklist for (bridged) tokens not tracked by CoinGecko. This is achieved by pruning tokens that are not tracked by Coingecko and where the _hide_ boolean is _true_ (which typically happens with bridged tokens, tokens with no supply or volume, ...).

## Checklist

- [x] (optional) As a contributor, my Ethereum address/ENS is: ens.rigoblock.eth
- [x] (optional) As a contributor, my Twitter handle is: @gab_rigo

## How to test?
check the current rigoblock app: the full token list is returned for each pool. Run this branch locally, you'll see only tokens with positive balances are returned.
